project/src/ui/hud/SettingPane.gd

```diff
@@ -109,7 +126,24 @@ func _change_profile_name(text: String) -> void:
 func _on_world_selected(index: int) -> void:
 	master_manager.active_profile.environment = world_list.get_item_text(index)
 	master_manager.load_profile(master_manager.active_profile)
+	
+
+func _select_compiler() -> void:
+	compiler_btn.get_popup().clear()
+	compilers = Toolchain.new().find_compilers()
+	for compiler in compilers:
+		compiler_btn.get_popup().add_item(compiler.name + " (" + compiler.version + ")")
 
+	
+func _on_compiler_selected(index: int) -> void:
+	var newCompiler = compilers[index]
+	var compilerProfile = master_manager.active_profile.compiler
+	compilerProfile.name = newCompiler.name
+	compilerProfile.path = newCompiler.path
+	compilerProfile.version = newCompiler.version
+	sketch_manager.set_compiler(compilerProfile)
+	_save_profile()
```

Comment:
shoudn't we only save when we click save?

Reply:
The problem with this is that I already call set_compiler in the sketch_manager when I select a compiler. But when the user does not save then, it could lead to a wrong state. I could also call set_compiler when saving and see whether the compiler has changed, if you like that more.

Reply:
I removed the automatic save. I think it should actually be fine to set the compiler without saving that in the profile.

Reply:
Saving means saving it to the disk, you should still apply the changes, just not save it. E.g. when the user presses reload it should revert back to the original profile that is saved to the disk.